src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -678,22 +678,35 @@ export class TerminalInstance implements ITerminalInstance {
 
 	public preparePathForTerminalAsync(path: string): Promise<string> {
 		return new Promise<string>(c => {
-			const hasSpace = path.indexOf(' ') !== -1;
-			if (platform.isWindows) {
-				const exe = this.shellLaunchConfig.executable;
-				if (!exe) {
+			const exe = this.shellLaunchConfig.executable;
+			if (!exe) {
+				if (platform.isWindows) {
 					c(path);
-					return;
+				} else {
+					c(this._escapeNonWindowsPath(path));
 				}
+				return;
+			}
+
+			const hasSpace = path.indexOf(' ') !== -1;
+			const isPowerShell = exe.indexOf('pwsh') !== -1 ||
+				this.title.indexOf('pwsh') !== -1 ||
+				exe.toLocaleLowerCase().indexOf('powershell') !== -1 ||
```

Comment:
`toLowerCase()` is probably better here

Reply:
actually I shouldn't need this here. The exe should always be lowercase.